### ISSUE TITLE
Libfabric dependency for fles_libfabric: missing minimal version

### ISSUE THREAD

**Issue by PALoizeau** [cbm-fles/flesnet#78]
Starting from (probably) 481f11d2219b3c2c01c079d36fe6648b94c8c683, compiling the master under Debian 10 fails with the following error in fles_libfabric when the libfabric package is installed:
```
[ 47%] Building CXX object lib/fles_libfabric/CMakeFiles/fles_libfabric.dir/ComputeNodeConnection.cpp.o
In file included from /lustre/cbm/users/ploizeau/mcbm2021/flesnet/lib/fles_libfabric/providers/LibfabricBarrier.hpp:8,
                 from /lustre/cbm/users/ploizeau/mcbm2021/flesnet/lib/fles_libfabric/Connection.hpp:11,
                 from /lustre/cbm/users/ploizeau/mcbm2021/flesnet/lib/fles_libfabric/ComputeNodeConnection.hpp:8,
                 from /lustre/cbm/users/ploizeau/mcbm2021/flesnet/lib/fles_libfabric/ComputeNodeConnection.cpp:4:
/lustre/cbm/users/ploizeau/mcbm2021/flesnet/lib/fles_libfabric/providers/LibfabricCollective.hpp:17:10: fatal error: rdma/fi_collective.h: No such file or directory
 #include <rdma/fi_collective.h>
          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [lib/fles_libfabric/CMakeFiles/fles_libfabric.dir/build.make:63: lib/fles_libfabric/CMakeFiles/fles_libfabric.dir/ComputeNodeConnection.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:679: lib/fles_libfabric/CMakeFiles/fles_libfabric.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

When trying to track the file in the [libfabric repository](https://github.com/ofiwg/libfabric/commits/master/include/rdma/fi_collective.h), I came to the conclusion that it was introduced in v1.8.0 (bottom of the following release [page](https://github.com/ofiwg/libfabric/releases?after=v1.9.1rc2)), while Debian 10 ships 1.6.2-3.

Unfortunately our current CMake config checks only if libfabric is present without checking the version.

I think we could 
-  Introduce a version check at https://github.com/cbm-fles/flesnet/blob/92ff50ead204d0acb4fccd9cbb9876817d077528/CMakeLists.txt#L44
- Add a related note in INSTALL.md in the `Build Dependencies` section

**Comment by oTTer-Chief** [comment]
Hi Pierre,

sry you had to debug this. Came to my attention yesterday and we are already discussing how to handle it.

For the time being the best solution to get it running my be to uninstall the libfabric-dev package. In that case the find_packge fails and it will build without libfabric.

**Comment by PALoizeau** [comment]
Should we consider that it was fixed by #79 and close the issue?

Or do you want to keep it open for when we enable it again to remember to introduce a minimal version?

**Comment by oTTer-Chief** [comment]
I'll close it for now
